electionleaflets/apps/leaflets/managers.py

```diff
@@ -0,0 +1,161 @@
+                                ),
+                                ARRAY[]::TEXT[]
+                            )
+                        ) AS name
+                    ) AS unique_names
+                ),
+                ''::tsvector
+            ) AS name_search_vector
+        FROM leaflets_leaflet ll
+    ) AS subquery
+    WHERE leaflets_leaflet.id = subquery.id;
+"""
+
+NAME_SEARCH_TRIGGER_SQL = """
+    DROP FUNCTION IF EXISTS leaflet_person_name_search_trigger() CASCADE;
+    CREATE FUNCTION leaflet_person_name_search_trigger() RETURNS trigger AS $$
+    DECLARE
+        json_name TEXT;
+    BEGIN
+        json_name := (
+            SELECT people_data->'person'->>'name'
+            FROM JSONB_EACH(new.people) AS people_entry(key, people_data)
+            LIMIT 1
+        );
+    
+        new.name_search_vector := (
+            SELECT
+            COALESCE(
+                (
+                    SELECT
+                        STRING_AGG(
+                            (
+                                SETWEIGHT(TO_TSVECTOR('simple', SPLIT_PART(name, ' ', 1)), 'B') ||
+                                SETWEIGHT(TO_TSVECTOR('simple', REGEXP_REPLACE(name, '^.* ', '')), 'A') ||
+                                SETWEIGHT(TO_TSVECTOR('simple', name), 'C')
+                            )::TEXT,
+                            ' '
+                        )::TSVECTOR
+                    FROM (
+                        SELECT DISTINCT UNNEST(
+                            COALESCE(
+                                ARRAY_APPEND(
+                                    (
+                                        SELECT ARRAY_AGG(people_data->'person'->>'name')
+                                        FROM JSONB_EACH(new.people) AS people_entry(key, people_data)
+                                        WHERE people_data->'person'->>'name' IS NOT NULL
+                                    ),
+                                    new.ynr_person_name
+                                ),
+                                ARRAY[]::TEXT[]
+                            )
+                        ) AS name
+                    ) AS unique_names
+                ),
+                ''::tsvector
+            )
+        );
+    
+        RETURN new;
+    END;
+    $$ LANGUAGE plpgsql;
+    DROP TRIGGER IF EXISTS tsvectorupdate ON leaflets_leaflet CASCADE;
+    CREATE TRIGGER tsvectorupdate BEFORE INSERT OR UPDATE
+        ON leaflets_leaflet FOR EACH ROW EXECUTE PROCEDURE leaflet_person_name_search_trigger();
+"""
+
+
+class LeafletQuerySet(models.query.QuerySet):
+    def _run_sql(self, SQL):
+        with connection.cursor() as cursor:
+            cursor.execute(SQL)
+
+    def update_name_search(self):
+        self._run_sql(POPULATE_NAME_SEARCH_COLUMN_SQL)
+
+    def update_name_search_trigger(self):
+        self._run_sql(NAME_SEARCH_TRIGGER_SQL)
+
+    def search_person_by_name(self, name: str) -> "models.query.QuerySet":
```

Comment:
Again, there's a couple of bits in here that could benefit from a comment or two explaining the 'why's